### PR TITLE
unit test baseline과 서비스 계층 테스트 도입

### LINKED ISSUE #86: 백엔드 unit test baseline과 서비스 계층 테스트 도입
## 대상 저장소
`alexization/git-ranker`

## 문제 / 배경
`git-ranker` 백엔드는 현재 `src/test/` tree가 비어 있고 `build.gradle`에 test runtime이 없어, backend 변경이 `./gradlew build`만으로 검증되고 있다.
이 상태에서는 핵심 비즈니스 service 로직의 회귀를 물리적 품질 게이트에서 잡기 어렵고, pure component 및 service 계층의 contract도 코드만으로 남아 있다.

## 왜 지금 필요한가
backend unit test를 다시 baseline으로 복구하고 `./gradlew test`를 필수 gate로 세워야 이후 기능 작업과 리팩터링에서 회귀를 빠르게 감지할 수 있다.

## 기대 결과
- `./gradlew test`와 `./gradlew build`가 backend 기본 verification command set으로 정렬된다.
- pure component와 domain service 계층의 핵심 성공/실패/경계 branch가 unit test로 고정된다.
- CI / deploy pre-check가 unit-test gate를 통과한 변경만 publish하도록 정렬된다.

## 범위
- `build.gradle` test runtime 및 JUnit platform baseline 정렬
- `.github/workflows/ci.yml`, `.github/workflows/deploy.yml` verification lane 정렬
- `AGENTS.md` verification entrypoint 문서화
- `src/test/**`에 pure component 및 domain service unit test 추가
- locale/time-dependent 테스트 리스크를 줄이기 위한 최소 production seam 보강

## 비범위
- integration test / Testcontainers lane 복구
- controller, security, batch, config, infrastructure 계층 테스트
- API / DTO / schema / DB migration 변경

## 접근 메모
- 테스트는 단순 build gate가 아니라 비즈니스 contract 검증을 우선한다.
- gate는 `./gradlew test`로 두고, packaging build는 workflow에서 `./gradlew build -x test`로 분리해 중복 실행을 줄인다.
- 코드나 문서만으로 도메인 의도가 잠기지 않는 부분은 의미 있는 branch/edge case 위주로 test를 작성한다.

## 영향 / 의존성
- backend verification lane이 `build-only`에서 `test + build`로 변경된다.
- 기존 배포 workflow의 verify 단계 수행 시간이 달라질 수 있다.

## 리스크 / 오픈 포인트
- 일부 service test의 cookie/header, response mapping assertion 깊이는 아직 균일하지 않다.
- future에 `check`에 추가 검증이 연결되면 workflow의 `build -x test` 사용 여부를 다시 점검해야 한다.

### PR DESCRIPTION
**Pull request by alexization**
## 1) 요약
- 무엇이 변경되었나요?
  - backend unit test baseline을 복구하고 `src/test/**`에 pure component + domain service 계층 unit test를 추가했습니다.
  - CI / deploy verify 단계에 `./gradlew test`를 넣고 packaging build는 `./gradlew build -x test`로 분리했습니다.
  - locale / wall-clock 의존을 줄이기 위해 badge formatting과 일부 time-dependent path에 최소 seam을 추가했습니다.
- 왜 지금 필요한가요?
  - 현재 backend는 `build-only` 검증이라 비즈니스 service 회귀를 물리적 gate에서 잡지 못했습니다.

## 2) 연관 이슈
- Closes #86
- 관련 frontend 이슈/PR: 없음

## 3) 문제와 목표
- 문제:
  - `git-ranker`는 test runtime과 `src/test/` tree가 없어 backend 변경이 `./gradlew build`만으로 검증되고 있었습니다.
- 사용자/운영자 관점의 결과:
  - pure component와 핵심 domain service branch가 unit test로 고정되고, verify 단계가 회귀를 더 일찍 잡습니다.
- 비목표:
  - integration test / Testcontainers lane 복구
  - controller / security / batch / infrastructure 계층 테스트 추가

## 4) 영향 범위
- 변경된 패키지/모듈:
  - `domain.badge`
  - `domain.auth.service`
  - `domain.log`
  - `domain.ranking`
  - `domain.user.service`
  - `global.util`
  - `src/test/**`
  - `build.gradle`, `.github/workflows/*`, `AGENTS.md`
- API/DTO/Schema 영향:
  - 없음
- DB/Cache/Batch/Scheduler 영향:
  - runtime schema 변경 없음
  - verify workflow에서 unit-test gate가 추가됨
- 보안/권한 영향:
  - 없음

## 5) 검증 증거

| 유형 | 명령어 / 증거 | 결과 |
| --- | --- | --- |
| Build | `./gradlew build` | 통과 |
| Unit | `./gradlew test` | 통과 |
| Integration | `미실행(이번 spec 범위에서 제외)` | 생략 |
| Coverage | `미실행(coverage gate는 이번 spec 비범위)` | 생략 |
| API/Manual Smoke | `미실행(unit-test baseline 작업)` | 생략 |

## 6) 관측성 확인
- 확인한 로그:
  - unit-test baseline 작업이라 별도 runtime log 확인 없음
- 확인한 메트릭:
  - 없음
- 확인한 trace/dashboard/query:
  - 없음

## 7) AI 리뷰 메모 (선택)
- Codex:
  - latest verification 기준 `./gradlew test`, `./gradlew build` 통과
- CodeRabbitAI:
  - 없음

## 8) 리스크 및 롤백
- 리스크:
  - 일부 service 영역의 cookie/header, response mapping assertion 깊이는 여전히 균일하지 않습니다.
  - future에 `check`에 새 검증이 묶이면 workflow의 `build -x test` 사용 여부를 다시 점검해야 합니다.
- 롤백 계획:
  - test baseline 도입 전 상태로 `build.gradle`, workflow, `AGENTS.md`, `src/test/**`를 되돌립니다.

## 9) 체크리스트
- [x] 연관 이슈가 연결되어 있음
- [x] Build / Unit / Integration 결과가 기입되어 있음
- [x] API/스키마/배치 영향이 반영되었거나 없음을 명시함
- [x] 로그/메트릭/trace 확인 내용을 적었거나 불필요 사유를 적음
- [x] 문서 또는 후속 이슈가 업데이트되었거나 불필요 사유를 적음

## 릴리스 노트

* **버그 수정**
  * 배지 포맷팅에서 로케일별 숫자 형식 처리 개선으로 일관된 표시 보장

* **테스트**
  * 인증, 배지, 사용자, 순위 및 기타 핵심 기능에 대한 포괄적인 단위 테스트 추가

* **Chores**
  * CI/CD 워크플로우 최적화 (테스트 단계 분리)
  * 테스트 의존성 및 빌드 구성 업데이트

### CODE REVIEW

File: src/test/java/com/gitranker/api/domain/auth/service/AuthServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class AuthServiceTest {
+
+    private static final LocalDateTime VALID_EXPIRY = LocalDateTime.of(2030, 1, 1, 0, 0);
+    private static final LocalDateTime EXPIRED_AT = LocalDateTime.of(2020, 1, 1, 0, 0);
```
[the remark this answers is not in the document]
Comment (alexization) [reply src/test/java/com/gitranker/api/domain/auth/service/AuthServiceTest.java]: 반영했습니다. `AuthServiceTest`의 고정 만료 상수는 제거하고 상대 시각 helper(`validExpiry()`, `expiredAt()`)로 바꿨습니다. 최신 커밋 `9b99aaf` 기준으로 `./gradlew test`, `./gradlew build`도 다시 통과했습니다.

File: src/test/java/com/gitranker/api/domain/user/service/UserRegistrationServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        OAuthAttributes attributes = oauthAttributes("new-user", "new-user@example.com", "https://images.example.com/new-user.png");
+        ActivityStatistics totalStats = stats(10, 2, 3, 1, 4);
+        ActivityStatistics baselineStats = stats(5, 1, 1, 1, 2);
+        User savedUser = savedUser(1L, "new-user");
+        ActivityLog latestLog = emptyActivityLog(savedUser);
+
+        when(userRepository.findByNodeId("MDQ6VXNlcjEyMzQ1")).thenReturn(Optional.empty());
+        when(gitHubActivityService.fetchRawAllActivities("new-user", attributes.githubCreatedAt())).thenReturn(null);
+        when(gitHubDataMapper.toActivityStatistics(null)).thenReturn(totalStats);
+        when(baselineStatsCalculator.calculate(any(User.class), org.mockito.ArgumentMatchers.isNull())).thenReturn(baselineStats);
```
[the remark this answers is not in the document]
Comment (alexization) [reply src/test/java/com/gitranker/api/domain/user/service/UserRegistrationServiceTest.java]: 반영했습니다. 신규 사용자 등록 테스트에서 `null` raw response 대신 `GitHubAllActivitiesResponse.empty()`를 사용하도록 바꾸고, mapper와 baseline calculator stubbing도 같은 fixture로 맞췄습니다. 최신 커밋 `9b99aaf` 기준으로 검증까지 다시 통과했습니다.
